public/app/features/alerting/unified/Silences.tsx

```diff
@@ -1,59 +1,18 @@
-import React, { useCallback, useEffect } from 'react';
+import React from 'react';
 import { Route, RouteChildrenProps, Switch } from 'react-router-dom';
 
-import { Alert, withErrorBoundary } from '@grafana/ui';
-import { Silence } from 'app/plugins/datasource/alertmanager/types';
-import { useDispatch } from 'app/types';
+import { withErrorBoundary } from '@grafana/ui';
 
-import { featureDiscoveryApi } from './api/featureDiscoveryApi';
 import { AlertmanagerPageWrapper } from './components/AlertingPageWrapper';
 import { GrafanaAlertmanagerDeliveryWarning } from './components/GrafanaAlertmanagerDeliveryWarning';
 import SilencesEditor from './components/silences/SilencesEditor';
 import SilencesTable from './components/silences/SilencesTable';
 import { useSilenceNavData } from './hooks/useSilenceNavData';
-import { useUnifiedAlertingSelector } from './hooks/useUnifiedAlertingSelector';
 import { useAlertmanager } from './state/AlertmanagerContext';
-import { fetchAmAlertsAction, fetchSilencesAction } from './state/actions';
-import { SILENCES_POLL_INTERVAL_MS } from './utils/constants';
-import { AsyncRequestState, initialAsyncRequestState } from './utils/redux';
 
 const Silences = () => {
   const { selectedAlertmanager } = useAlertmanager();
 
-  const dispatch = useDispatch();
-  const silences = useUnifiedAlertingSelector((state) => state.silences);
-  const alertsRequests = useUnifiedAlertingSelector((state) => state.amAlerts);
-  const alertsRequest = selectedAlertmanager
-    ? alertsRequests[selectedAlertmanager] || initialAsyncRequestState
-    : undefined;
-
-  const { currentData: amFeatures } = featureDiscoveryApi.useDiscoverAmFeaturesQuery(
-    { amSourceName: selectedAlertmanager ?? '' },
-    { skip: !selectedAlertmanager }
-  );
-
-  useEffect(() => {
-    function fetchAll() {
-      if (selectedAlertmanager) {
-        dispatch(fetchSilencesAction(selectedAlertmanager));
-        dispatch(fetchAmAlertsAction(selectedAlertmanager));
-      }
-    }
-    fetchAll();
-    const interval = setInterval(() => fetchAll, SILENCES_POLL_INTERVAL_MS);
-    return () => {
-      clearInterval(interval);
-    };
```

Comment:
I don't think this was working before - I didn't see it refetch silences periodically

Reply:
You're right, it doesn't work. The function passed to `setInterval` returns `fetchAll` function instead of invoking it, so no requests are made

Reply:
Well now it works, and it does `refetchOnFocus` as well 🙌